### PR TITLE
feat: Add SSH publicKey auth support

### LINKED ISSUE #6800: Core: Support publicKey git authorization via SSH in resource-service
**User story**
As a user, I want to be able to use my publicKey for Keptn git repository authorization like I use it when working with remote git repository

**Technical details**
There is a need to store the publicKey into secret and use it when:
- clone
- pull
- push
- fetch

Also the gitRemoteUrl will use SSH protocol. The implementation of the feature (without tests and refactoring) is available as a part of this draft pull-requests: 
- https://github.com/keptn/keptn/pull/6746
- https://github.com/keptn/go-utils/pull/384

Research ticket about this feature: https://github.com/keptn/keptn/issues/6647

**Acceptance Criteria**
- [ ] Implement support of publicKey git authorization via SSH in resource-service
- [ ] Implement acceptation of new parameters in CLI
- [ ] Add new parameters to API endpoints in shipyard-controller
- [ ] Update go-utils accordingly

**DoD**
- [ ] Acceptance Criteria are implemented
- [ ] Unit tests
- [ ] Integration test

### PR DESCRIPTION
**Pull request by odubajDT**
Signed-off-by: odubajDT <ondrej.dubaj@dynatrace.com>

## This PR

- adds SSH publicKey auth support

### Related Issues

Fixes #6800 

### Integration Tests
https://github.com/keptn/keptn/actions/runs/1857178254 and https://github.com/keptn/keptn/actions/runs/1887111092
Note: integration test checking the new functionality failed but only due to timeout, already fixed.

### CODE REVIEW

File: cli/cmd/create_project.go [hunk 1]
```diff
@@ -18,19 +20,27 @@ import (
...
 You can configure a Git upstream repository using: 
 
-keptn update project PROJECTNAME --git-user=GIT_USER --git-token=GIT_TOKEN --git-remote-url=GIT_REMOTE_URL
+keptn update project PROJECTNAME --git-user=GIT_USER --git-remote-url=GIT_REMOTE_URL --git-token=GIT_TOKEN
+
+or (only for resource-service)
+
+keptn update project PROJECTNAME --git-user=GIT_USER --git-remote-url=GIT_REMOTE_URL --git-private-key=PRIVATE_KEY_PATH --git-private-key-pass=PRIVATE_KEY_PASSPHRASE
+
+Please be aware that authentification with public/private key is supported only when using resource-service.
```
Comment (warber) [inline cli/cmd/create_project.go:43]: "authentication"

File: cli/cmd/create_project.go [hunk 2]
```diff
@@ -41,12 +51,19 @@ var crProjectCmd = &cobra.Command{
 The shipyard file describes the used stages. These stages are defined by name, as well as their task sequences.
 
 By executing the *create project* command, Keptn initializes an internal Git repository that is used to maintain all project-related resources. 
-To upstream this internal Git repository to a remote repository, the Git user (*--git-user*), an access token (*--git-token*), and the remote URL (*--git-remote-url*) are required.
+To upstream this internal Git repository to a remote repository, the Git user (*--git-user*) and the remote URL (*--git-remote-url*) are required
+together with private key (*--git-private-key*) or access token (*--git-token*). Please be aware that authentification with public/private key is
```
Comment (warber) [inline cli/cmd/create_project.go:55]: authentication

File: cli/cmd/update_project.go [hunk 3]
```diff
@@ -31,11 +35,17 @@ var upProjectCmd = &cobra.Command{
 Updating a shipyard file is not possible.
 
 By executing the update project command, Keptn will add the provided upstream repository to the existing internal Git repository that is used to maintain all project-related resources. 
-To upstream this internal Git repository to a remote repository, the Git user (--git-user), an access token (--git-token), and the remote URL (--git-remote-url) are required.
+To upstream this internal Git repository to a remote repository, the Git user (--git-user) and the remote URL (*--git-remote-url*) are required
+together with private key (*--git-private-key*) or access token (*--git-token*). Please be aware that authentification with public/private key is
```
Comment (warber) [inline cli/cmd/update_project.go:39]: authentication

**Review by warber — CHANGES_REQUESTED** [review]

File: test/go-tests/test_utils.go [hunk 4]
```diff
@@ -169,7 +170,7 @@ func GetInternalKeptnAPI(ctx context.Context, internalService, localPort string,
 	return keptnInternalAPI, nil
 }
 
-func CreateProject(projectName string, shipyardFilePath string, recreateIfAlreadyThere bool) (string, error) {
+func CreateProject(projectName string, shipyardFilePath string, recreateIfAlreadyThere bool, useSSH bool) (string, error) {
```
[the remark this answers is not in the document]
Comment (odubajDT) [reply test/go-tests/test_utils.go]: fixed, thx

**Review by RealAnna — COMMENTED** [review]
LGTM

File: test/go-tests/test_utils.go [hunk 5]
```diff
@@ -225,6 +228,93 @@ func CreateProject(projectName string, shipyardFilePath string, recreateIfAlread
 	return "", err
 }
 
+func CreateProjectWithSSH(projectName string, shipyardFilePath string, recreateIfAlreadyThere bool) (string, error) {
```
Comment (warber) [inline test/go-tests/test_utils.go:231]: Quite some stuff in `CreateProject` and `CreateProjectWithSSH` are identical - pls refactor

**Review by warber — CHANGES_REQUESTED** [review]

File: test/go-tests/test_utils.go (hunk 5 above)
Comment (odubajDT) [reply test/go-tests/test_utils.go:243]: done

**Review by warber — APPROVED** [review]

**Pull request merged**